plugins/inputs/traefik/traefik.go

```diff
@@ -0,0 +1,169 @@
+
+    ## Optional TLS Config
+    # tls_ca = "/etc/telegraf/ca.pem"
+    # tls_cert = "/etc/telegraf/cert.pem"
+    # tls_key = "/etc/telegraf/key.pem"
+    ## Use TLS but skip chain & host verification
+    # insecure_skip_verify = false
+
+    # Additional tags
+    [inputs.traefik.tags]
+      instance = "prod"
+`
+
+func (t *Traefik) Description() string {
+	return "Gather health check status from services registered in Traefik"
+}
+
+func (t *Traefik) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Traefik) submitStatusCodeMeasurement(acc telegraf.Accumulator, check *HealthCheck, tags map[string]string, fields map[string]interface{}) error {
+	fields["total_count"] = check.TotalCount
+	fields["uptime_sec"] = check.UptimeSec
+	fields["unixtime"] = check.Unixtime
+
+	for key, value := range check.TotalStatusCodeCount {
+		newTags := copyTags(tags)
+		newFields := copyFields(fields)
+		newTags["status_code"] = key
+		newFields["count"] = value
+		acc.AddFields("traefik_status_codes", newFields, newTags)
+	}
+
+	return nil
+}
+
+func copyFields(m map[string]interface{}) map[string]interface{} {
+	fields := make(map[string]interface{})
+	for k, v := range m {
+		fields[k] = v
+	}
+	return fields
+}
+
+func copyTags(m map[string]string) map[string]string {
+	tags := make(map[string]string)
+	for k, v := range m {
+		tags[k] = v
+	}
+	return tags
+}
+
+func (t *Traefik) submitPrimaryMeasurement(acc telegraf.Accumulator, check *HealthCheck, tags map[string]string, fields map[string]interface{}) error {
+	newTags := copyTags(tags)
+	newFields := copyFields(fields)
+
+	for key, value := range check.TotalStatusCodeCount {
+		newFields[fmt.Sprintf("status_code_%v", key)] = value
+	}
+
+	newFields["total_response_time_sec"] = check.TotalResponseTimeSec
+	newFields["average_response_time_sec"] = check.AverageResponseTimeSec
+	newFields["total_count"] = check.TotalCount
+	newFields["uptime_sec"] = check.UptimeSec
+	newFields["unixtime"] = check.Unixtime
+
+	acc.AddFields("traefik", newFields, newTags)
+	return nil
+}
+
+func (t *Traefik) createHttpClient() (*http.Client, error) {
+	tlsCfg, err := t.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: t.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (t *Traefik) Gather(acc telegraf.Accumulator) error {
+	if t.client == nil {
+		client, err := t.createHttpClient()
+		if err != nil {
+			return err
+		}
+		t.client = client
+	}
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%v/health", t.Address), nil)
+	if err != nil {
+		return err
+	}
+	start := time.Now()
+	resp, err := t.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	healthData := &HealthCheck{}
+	json.NewDecoder(resp.Body).Decode(&healthData)
+	t.lastRequestTiming = time.Since(start).Seconds()
+
+	tags := map[string]string{"server": t.Address}
```

Comment:
Lets call the tag `source` (rather than `server` (4413)) and only use the parsed url's `.Hostname()` as the value. Also, save this as a variable on the struct so it doesn't have to be computed each call to `Gather`.